k2/csrc/fsa_equivalent.cc

```diff
@@ -199,6 +202,12 @@ bool IsRandEquivalent(const Fsa &a, const float *a_weights, const Fsa &b,
        (*(labels_difference.begin())) != kEpsilon))
     return false;
 
+  double best_distance = kDoubleNegativeInfinity;
+  if (beam != kFloatInfinity) {
+    best_distance =
+        ShortestDistance<Type>(valid_a, valid_a_weights.data()) - beam;
+  }
```

Comment:
I prefer if you set it to -infinity in the `else` statement of this, and just not initialize it when you declare it.  Would be clearer.

Reply:
Ok